sim_pipeline/halos/halos_lens.py

```diff
@@ -0,0 +1,233 @@
+    """
+    c_200 = A * ((1 + z) ** d) * (mass ** m)
+    c_200 = np.maximum(c_200, 1)
+    return c_200
+    # TODO: Add test function
+
+
+class HalosLens(object):
+    """
+    A class to manage the lensing properties of the halos.
+
+    Parameters
+    ----------
+    halos_list : table
+        List of halos with ['z']&['mass'] for which the lensing properties are computed.
+    cosmo : astropy.cosmology instance, optional
+        The cosmology assumed in the lensing computations. If none is provided, the default astropy cosmology is used.
+    sky_area : float, optional
+        The total sky area over which the halos are distributed. Default is the full sky (4pi steradians).
+
+    Attributes
+    ----------
+    halos_list : table
+        List of halos for which the lensing properties are computed.
+    n_halos : int
+        The number of halos in `halos_list`.
+    sky_area : float
+        The total sky area in deg^2 over which the halos are distributed.
+    redshift_list : array_like
+        The redshifts of the halos.
+    mass_list : array_like
+        The masses of the halos.
+    cosmo : astropy.Cosmology instance
+        The cosmology assumed in the lensing computations.
+    lens_model : lenstronomy.LensModel instance
+                The LensModel object with a NFW profile for every halo.
+    Methods
+    -------
+    random_position() :
+        Returns random x and y coordinates in the sky, uniformly distributed within a circular sky area.
+    get_nfw_kwargs() :
+        Returns the scale radius, observed bending angle at the scale radius, and positions of the halos
+        in the lens plane.
+    get_halos_lens_kwargs() :
+        Returns the list of keyword arguments for each halo to be used in the lens model.
+    get_convergence_shear() :
+        Returns the convergence and shear at the origin due to all the halos.
+    """
+
+    # TODO: ADD test functions
+    def __init__(self, halos_list, cosmo=None, sky_area=4 * np.pi, samples_number=1000):
+        self.halos_list = halos_list
+        self.n_halos = len(self.halos_list)
+        self.sky_area = sky_area
+        self.redshift_list = halos_list['z']
+        self.mass_list = halos_list['mass']
+        self.samples_number = samples_number
+        if cosmo is None:
+            warnings.warn("No cosmology provided, instead uses astropy.cosmology import default_cosmology")
+            from astropy.cosmology import default_cosmology
+            self.cosmo = default_cosmology.get()
+        else:
+            self.cosmo = cosmo
+
+        self.lens_model = LensModel(lens_model_list=['NFW'] * self.n_halos,
+                                    lens_redshift_list=self.redshift_list,
+                                    cosmo=self.cosmo,
+                                    observed_convention_index=[],
+                                    multi_plane=True,
+                                    z_source=5
```

Comment:
the source redshift is hard-coded here. Perhaps have it as an input argument and make sure the halo rendering and LensModel instance class are consistent with each other